### DISCUSSION TITLE
Why can't I specify MIDI events in ODF?

### DISCUSSION THREAD

**Discussion by William-Purvis** [GrandOrgue/grandorgue#1735] [General]
I am currently working on a 3-manual organ with 64 tab stops and two dozen pistons. GrandOrgue keeps losing my settings
 and entering them via the GUI is very tedious. Why can't I specify them in the ODF, I'd be happy to write them in once
and if anything needs to be change simply update the ODF? I know that these are saved in the settinsg file , but there's a
massive amount of other stuff in there, most of which means nothing to me (what are all tose Frames?). I can extract
the relevant bits into a subset of setting and import that, but it would make more sense to have it all in my ODF.

**Comment by larspalo** [comment]
Because such settings have no place in the ODF. The ODF is for describing the pipe organ in question that is modeled, not any specific MIDI setup that you have on your console. Also, remember that the .organ file is supposed to be possible to use for any user with any kind of MIDI equipment at hand. Is that clear enough as answer to why you cannot specify MIDI events in the ODF?

For legacy reasons the following options are documented in the help:
```
StopControlMIDIKeyNumber
(integer 0 - 127, default: no MIDI event specified) Only used for building the initial configuration during the first load - provided just for HW1 compatibility. DEPRECATED.

MIDIProgramChangeNumber
(integer 1 - 128, default: no MIDI event specified) Only used for building the initial configuration during the first load - provided just for HW1 compatibility. DEPRECATED.
```

They are **not** meant to be used, as they are deprecated!

On a side note, as a friendly advice, (especially to all of you who are creating sample sets, or just creates a custom organ): if you're still working on an organ (meaning you're changing stuff in it, or intend to change things any further) don't create custom MIDI configurations for it yet. "Save" that until the organ is finished. The presets of GO are there to serve as convenience for the **user** of an end product (not the developer). So, don't _Save_ an organ (in GrandOrgue) that is still work in progress! It's best to only use the initial MIDI configurations until the organ is finished.

As this also extends to all voicing adjustments done in the GO GUI, wait with all such things until the organ is "finished". When the organ is finalized it's safe to create the settings file. Even though I have implemented a merging of voicing information from a .cmb to the .organ in GOODF, it's still necessary to be careful and not make any more changes after the .cmb is created to be sure that information is imported correctly.
